M/mlpack/build_tarballs.jl

```diff
@@ -0,0 +1,218 @@
+    # be available in the build environment...
+    mkdir build
+    cd build
+    echo "#!/bin/bash" > julia
+    echo "echo \"Fake Julia version 1.3.0\"" >> julia
+    chmod +x julia
+
+    if [[ $target == *apple* ]]
+    then
+        cmake \
+            -DDEBUG=OFF \
+            -DPROFILE=OFF \
+            -DUSE_OPENMP=OFF \
+            -DCMAKE_SYSTEM_NAME="Darwin" \
+            -DBoost_NO_BOOST_CMAKE=1 \
+            -DBUILD_JULIA_BINDINGS=ON \
+            -DJULIA_EXECUTABLE=`pwd`/julia \
+            -DBUILD_CLI_EXECUTABLES=OFF \
+            -DBUILD_TESTS=OFF \
+            -DARMADILLO_INCLUDE_DIR=${WORKSPACE}/srcdir/armadillo-9.800.3/tmp/include/ \
+            -DCMAKE_INSTALL_PREFIX=$prefix \
+            ..
+    elif [[ $target == *mingw* ]]
+    then
+        # We need to remove bigobj support, but we also need to force the linker
+        # to export all symbols because Windows DLLs have the concept of
+        # "exported" and "not exported" functions.
+        sed -i 's/-Wa,-mbig-obj/-Wl,--export-all-symbols/' ../CMakeLists.txt
+
+        cmake \
+            -DDEBUG=OFF \
+            -DPROFILE=OFF \
+            -DUSE_OPENMP=OFF \
+            -DBoost_NO_BOOST_CMAKE=1 \
+            -DBUILD_SHARED_LIBS=ON \
+            -DBUILD_JULIA_BINDINGS=ON \
+            -DJULIA_EXECUTABLE=`pwd`/julia \
+            -DBUILD_CLI_EXECUTABLES=OFF \
+            -DBUILD_TESTS=OFF \
+            -DARMADILLO_INCLUDE_DIR=${WORKSPACE}/srcdir/armadillo-9.800.3/tmp/include/ \
+            -DLAPACK_LIBRARY=$openblas_lib \
+            -DBLAS_LIBRARY=$openblas_lib \
+            -DCMAKE_INSTALL_PREFIX=$prefix \
+            -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+            ..
+    else
+        cmake \
+            -DDEBUG=OFF \
+            -DPROFILE=OFF \
+            -DUSE_OPENMP=OFF \
+            -DBoost_NO_BOOST_CMAKE=1 \
+            -DBUILD_JULIA_BINDINGS=ON \
+            -DJULIA_EXECUTABLE=`pwd`/julia \
+            -DBUILD_CLI_EXECUTABLES=OFF \
+            -DBUILD_TESTS=OFF \
+            -DARMADILLO_INCLUDE_DIR=${WORKSPACE}/srcdir/armadillo-9.800.3/tmp/include/ \
+            -DCMAKE_INSTALL_PREFIX=$prefix \
+            -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+            ..
+    fi
+    make -j2
+    make install
+
+    libdir=`find $prefix -iname 'lib*' -type d -maxdepth 1 | head -1`;
+    if [[ $target == *apple* ]]
+    then
+        cp -v src/mlpack/bindings/julia/mlpack/src/*.dylib $libdir/
+    elif [[ $target == *mingw* ]]
+    then
+        cp -v libmlpack_julia*.dll.a $libdir/
+        cp -v libmlpack_julia*.dll $prefix/bin/
+    else
+        cp -v src/mlpack/bindings/julia/mlpack/src/*.so $libdir/
+    fi
+
+    ls $prefix
+    ls $libdir
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line.
+platforms = supported_platforms()
+# Seems like boost_jll doesn't exist for FreeBSD, so we have to remove it.
```

Comment:
You may be interested in #364, and maybe answering the question asked by @staticfloat

Reply:
Thanks for the link, I'll respond there.